projects/VitalRecord/VitalRecord_util.cs

```diff
@@ -94,6 +94,143 @@ protected void UpdateEntry(bool shouldExist, [CallerMemberName] string propertyN
+        /// <param name="code">the code to identify the type of Observation</param>
+        /// <param name="extensionURL">if present, specifies that the value should be set on an extension with the provided URL instead</param>
+        protected Dictionary<string, string> GetObservationValue(string code, string extensionURL = null)
+        {
+            var entry = Bundle.Entry.Where(e => e.Resource is Observation obs && CodeableConceptToDict(obs.Code)["code"] == code).FirstOrDefault();
+            if (entry != null)
+            {
+                Observation observation = (Observation)entry.Resource;
+                if (extensionURL != null)
+                {
+                    Extension extension = observation?.Value?.Extension.FirstOrDefault(ext => ext.Url == extensionURL);
+                    if (extension != null && extension.Value != null && extension.Value.GetType() == typeof(CodeableConcept))
+                    {
+                        return CodeableConceptToDict((CodeableConcept)extension.Value);
+                    }
+                }
+                else
+                {
+                    return CodeableConceptToDict((CodeableConcept)observation.Value);
+                }
+            }
+            return EmptyCodeableDict();
+        }
+
+        /// <summary>Helper to support vital record property setter helper methods for values stored in Observations.</summary>
+        /// <param name="value">the coded value to set as the value of the property</param>
+        /// <param name="code">the code to specify the type of Observation</param>
+        /// <param name="codeSystem">the code system of the code specifying the type of Observation</param>
+        /// <param name="text">the text for the code specifying the type of Observation</param>
+        /// <param name="profileURL">the profile URL to include in the meta of the Observation</param>
+        /// <param name="section">the section of the composition the Observation should be added to</param>
+        /// <param name="extensionURL">if present, specifies that the value should be set on an extension with the provided URL instead</param>
+        /// <param name="propertyName">the name of the C# property, used to determine the subject ID</param>
+        protected void SetObservationValue(Dictionary<string, string> value, string code, string codeSystem, string text, string profileURL, string section, string extensionURL = null, [CallerMemberName] string propertyName = null)
+        {
+            var entry = Bundle.Entry.Where(e => e.Resource is Observation obs && CodeableConceptToDict(obs.Code)["code"] == code).FirstOrDefault();
+            Observation observation;
+            // If the observation is there we use it, otherwise create it
+            if (entry != null)
+            {
+                observation = (Observation)entry.Resource;
+            }
+            else
+            {
+                observation = new Observation();
```

Comment:
Do we still need to define the observations as fields and create methods file so we can refer to them in RestoreReferences (and maybe other places I'm not thinking of right now)? It looks like with this approach there wouldn't be a way to reference an observation like CodedRaceAndEthnicityObs since the observation is created on the spot. Maybe we need to try this out with RestoreReferences and see if things can still work.

Reply:
For the observations that this approach supports there aren't actually any references to restore, since to read the value of one of these observations or update an existing value we search for it directly in the bundle by the observation code and pull it from there, and the observation doesn't need to be referenced other than to set it or update it or get data from it.

If it winds up in a coded content bundle we will need to handle the observations by code in GetDemographicCodedContentBundle; I added a comment to that effect.